datafusion/common/src/pyarrow.rs

```diff
@@ -17,6 +17,10 @@
 
 //! Conversions between PyArrow and DataFusion types
 
+// TODO update to pyo3 new APIs
```

Comment:
I am going to need help upgrading the python API (or need to spend time learning about the python type system) https://pyo3.rs/v0.23.0/migration

Reply:
@timsaucer  hooked me up https://github.com/alamb/datafusion/pull/21